traits/trait_list_object.py

```diff
@@ -59,34 +59,117 @@ def __repr__(self):
 
 
 def _normalize_index(index, length):
-    """ Normalize integer index to range 0 to len (inclusive). """
+    """ Normalize integer index to range 0 to len (inclusive).
+
+    These rules reflect the normalization that Python uses for the
+    ``insert`` and ``pop`` operations.
+
+    """
+
     index = operator.index(index)
     if index < 0:
         return max(0, length + index)
     else:
         return min(length, index)
 
 
-def _normalize_slice(index, length):
-    """ Normalize slice start and stop to range 0 to len (inclusive). """
+def _normalize_slice_or_index(index, length):
+    """ Normalize a slice or index for use with __delitem__ or __setitem__.
 
-    # Non-extended slice (step = 1): return only need the start index.
-    if index.step is None or index.step == 1:
-        if index.start is None:
-            return 0
-        else:
-            return _normalize_index(index.start, length)
+    For slices with positive step, returns a slice that's equivalent for the
```

Comment:
Replace "returns a slice" with "returns a slice or index"; similarly for the next sentence about negative step.

Reply:
Done.